edk2toollib/uefi/edk2/path_utilities.py

```diff
@@ -56,25 +56,31 @@ def __init__(self, ws, packagepathlist):
             raise Exception("Invalid package path directory(s)")
 
     def GetEdk2RelativePathFromAbsolutePath(self, abspath):
+        ''' Given an absolute path return a edk2 path relative
+        to workspace or packagespath.
+
+        If not valid return None
+        '''
+
         relpath = None
         found = False
         if abspath is None:
             return None
-        for a in self.PackagePathList:
-            stripped = abspath.lower().partition(a.lower())[2]
-            if stripped:
-                # found our path...now lets correct for case
+        for a in (os.path.normcase(p) for p in self.PackagePathList):
+            if os.path.normcase(abspath).startswith(a):
```

Comment:
wouldn't it make more sense to convert abspath into normabspath or something rather than doing it each time?

Reply:
this isn't in a loop and will only do the conversion on the fly so it we break it won't even convert all of them.  I thought about doing this at the class level but it didn't seem right and it added complexity for no practical gain.  

If you look at GetContainingPackage i did convert them into a local list.  This is because it is called in a loop.